src/Storages/HDFS/StorageHDFS.cpp

```diff
@@ -712,6 +712,8 @@ void registerStorageHDFS(StorageFactory & factory)
         engine_args[0] = evaluateConstantExpressionOrIdentifierAsLiteral(engine_args[0], args.getLocalContext());
 
         String url = checkAndGetLiteralArgument<String>(engine_args[0], "url");
+        if (url.empty())
+            throw Exception("Storage HDFS requires valid URL to be set, empty string provided", ErrorCodes::BAD_ARGUMENTS);
```

Comment:
What will happen if I provide other invalid url like `' '` or `'42'` or `'/'`? Could you please add more negative tests?

Reply:
Looks like the root cause is here:
https://github.com/ClickHouse/ClickHouse/blob/78fc36ca498f37bb2199512f803852c66af6e4c1/src/Storages/HDFS/StorageHDFS.cpp#L123
`std::sting::find` may return `std::string::npos` and it's incorrect to add 2 to `npos` (it leads to overflow)

Reply:
oh, I did not notice your message, you was first, I race condition...